File: Floppy proyecto/app/src/main/java/com/example/floppy/data/Conexion/remote/ConnectionFirestore.java
package com.example.floppy.data.Conexion.remote;

import com.example.floppy.Callbacks.CallBackObjects;
import com.example.floppy.Callbacks.CallbackList;
import com.example.floppy.domain.remote.Interactor;
import com.example.floppy.domain.models.Chat;
import com.example.floppy.domain.models.Estado_User;
import com.example.floppy.domain.models.Message;
import com.example.floppy.domain.models.User;
import com.example.floppy.ui.login.LoginPresenter;
import com.example.floppy.ui.message.MessagePresenter;

import java.util.concurrent.CountDownLatch;

public interface ConnectionFirestore {

    void login(User user, CountDownLatch countDownLatch);

    void registerUserDataBase(User user, CountDownLatch countDownLatch);

    void updateUserDataBase(User user, CountDownLatch countDownLatch);

    void saveFoto(byte[] img, User user, CountDownLatch countDownLatch);

    void updateUser(User user, CountDownLatch countDownLatch);

    void getStates(CountDownLatch countDownLatch, String id);

    void getMyData(CountDownLatch countDownLatch, Interactor interactor, LoginPresenter presenter);

    void searchChat(Chat chat, CountDownLatch countDownLatch);

    void createChat(Chat chat, CountDownLatch countDownLatch);

    void getAllUsers(CountDownLatch countDownLatch);

    void getMessagesByIdChat(Chat chat, CallbackList<Message> callback, MessagePresenter presenter);

    void sendMessages(String idChat, String conversation,String date);

    void cancelListener();

    void updateState(String idUser, Estado_User estado_user);

}

File: Floppy proyecto/data/src/main/java/com/estarly/data/remote/ConnectionFirestore.kt
package com.estarly.data.remote

import java.util.concurrent.CountDownLatch

interface ConnectionFirestore {
    fun login(data: Array<String>, countDownLatch: CountDownLatch)

    fun registerUserDataBase(data: Array<String>, countDownLatch: CountDownLatch)

    fun updateUserDataBase(
        map: Map<String, Any>,
        idUser: String,
        countDownLatch: CountDownLatch
    )

//    void saveFoto(byte[] img, User user, CountDownLatch countDownLatch);
//
//    void updateUser(User user, CountDownLatch countDownLatch);

    //    void saveFoto(byte[] img, User user, CountDownLatch countDownLatch);
    //
    //    void updateUser(User user, CountDownLatch countDownLatch);
    fun getStates(countDownLatch: CountDownLatch, id: String)

    fun getMyData(countDownLatch: CountDownLatch, idUser: String)

    fun searchChat(data: Array<String>, countDownLatch: CountDownLatch)

    fun createChat(map: Map<String, Any>, countDownLatch: CountDownLatch)

    fun getAllUsers(countDownLatch: CountDownLatch, idUser: String)

    fun getMessagesByIdChat(idChat: String, idFriend: String,callbackChat: (String)->Unit,callbackStatus: (String)->Unit)

    fun sendMessages(idChat: String, conversation: String)

    fun cancelListener()

    fun updateState(idUser: String, estado_user: String)

    fun getFriends(countDownLatch: CountDownLatch, idFriend: String?)

    fun listenerStatusUser(idFriend: String?, callback: (String) -> Unit)

    fun listenerChatFriend(
        idChat: String,
        callback: (String)->Unit
    )

    fun downloadFile(nameFile: String, typeMessage: String)
}